### ISSUE TITLE
Too much external dependencies?

### ISSUE THREAD

**Issue by kron4eg** [prometheus/tsdb#127]
Here is single `import "github.com/prometheus/tsdb"` results too

```
$ dep status
PROJECT                                           CONSTRAINT     VERSION        REVISION  LATEST   PKGS USED
github.com/beorn7/perks                           *              branch master  4c0e845   4c0e845  1
github.com/cespare/xxhash                         *              v1.0.0         5c37fe3   5c37fe3  1
github.com/coreos/etcd                            *              v3.2.6         9d43462   9d43462  1
github.com/coreos/go-systemd                      *              v15            d219646   d219646  1
github.com/coreos/pkg                             *              v3             3ac0863   3ac0863  1
github.com/dgryski/go-bits                        *              branch master  2ad8d70   2ad8d70  1
github.com/go-kit/kit                             *              v0.5.0         a9ca672   a9ca672  1
github.com/go-logfmt/logfmt                       *              v0.3.0         390ab79   390ab79  1
github.com/go-stack/stack                         *              v1.6.0         817915b   817915b  1
github.com/golang/protobuf                        *              branch master  ab9f9a6   ab9f9a6  1
github.com/kr/logfmt                              *              branch master  b84e30a   b84e30a  1
github.com/matttproud/golang_protobuf_extensions  *              v1.0.0         3247c84   3247c84  1
github.com/nightlyone/lockfile                    *              branch master  6a197d5   6a197d5  1
github.com/oklog/ulid                             *              v0.3.0         d311cb4   d311cb4  1
github.com/pkg/errors                             *              v0.8.0         645ef00   645ef00  1
github.com/prometheus/client_golang               *              v0.8.0         c5b7fcc   c5b7fcc  1
github.com/prometheus/client_model                *              branch master  6f38060   6f38060  1
github.com/prometheus/common                      *              branch master  61f87aa   61f87aa  3
github.com/prometheus/procfs                      *              branch master  e645f4e   e645f4e  2
github.com/prometheus/tsdb                        branch master  branch master  79dc0ed   79dc0ed  3
golang.org/x/net                                  *              branch master  1c05540   1c05540  1
golang.org/x/sync                                 *              branch master  f52d181   f52d181  1
```

Most are just transitive dependencies from `github.com/coreos/etcd` and `github.com/go-kit/kit`.

Does it make sense to import etcd (and it's transitive dependencies) for just couple of functions like OpenDir and Preallocate? What do you think about "little copy" to replace "dependencies tree"?

**Comment by fabxc** [comment]
Yea, using etcd's fileutils was an easy means to get some reliable functionality in. 
It would probably be best to replace those with our own functions. Contributions would be welcome of course :)

**Comment by mdlayher** [comment]
Is it possible to just import the fileutil package with dep without bringing in the rest of the tree?  Looks like the package provides some decent value, particularly for darwin users.

**Comment by kron4eg** [comment]
@mdlayher yeah, in two steps. `dep ensure && dep prune`. Ensure will install everything, prune will remove unused sources that will not be used in build (except _test.go and non go files).

**Comment by Gouthamve** [comment]
Closed in #161
